admin/commands/management/remove.go

```diff
@@ -42,9 +45,53 @@ type removeMySQLCommand struct {
 	ServiceType string
 	ServiceName string
 	ServiceID   string
+	NodeID      string
 }
 
 func (cmd *removeMySQLCommand) Run() (commands.Result, error) {
+	if cmd.NodeID == "" {
+		status, err := agentlocal.GetStatus(agentlocal.DoNotRequestNetworkInfo)
+		if err != nil {
+			return nil, err
+		}
+		cmd.NodeID = status.NodeID
+	}
+
+	if cmd.ServiceID == "" && cmd.ServiceName == "" {
+		// PMM-10095: Automatic service lookup during removal
+		//
+		// Get services and remove it automatically once it's only one
+		// service registered
+		servicesRes, err := inventoryClient.Default.Services.ListServices(&services.ListServicesParams{
+			Body: services.ListServicesBody{
+				NodeID:      cmd.NodeID,
+				ServiceType: cmd.serviceType(),
```

Comment:
@BupycHuk returns services filtered by node_id and serviceType. In that case, we are sure that we'll have only one instance per service type.

Reply:
Yeah, you are right. Missed that